Assets/MixedRealityToolkit-Examples/UX/Scripts/ProgressExamples.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using System.Collections;
+using MixedRealityToolkit.UX.Progress;
+using MixedRealityToolkit.InputModule;
+using MixedRealityToolkit.InputModule.InputSources;
+using UnityEngine.XR.WSA.Input;
+using MixedRealityToolkit.Utilities.Solvers;
+using UnityEngine.XR.WSA;
+
+namespace MixedRealityToolkit.Examples.UX
+{
+    public class ProgressExamples : MonoBehaviour
+    {
+        public GameObject SceneObject;
+
+        [Header("How long to spend on each stage of loading")]
+        public float LeadInTime = 1.5f;
+        public float LoadingTime = 5f;
+        public float FinishTime = 1.5f;
+
+        [Header("Set these to override the defaults set in the ProgressIndicator prefab")]
+        public GameObject LoadingPrefab = null;
+        public Texture2D LoadingIcon = null;
+
+        [Header("Messages displayed during loading")]
+        public string LeadInMessage = "Lead in Message";
+        public string LoadTextMessage = "Loading with message only";
+        public string LoadOrbsMessage = "Loading with Orbs";
+        public string LoadIconMessage = "Loading with Icon";
+        public string LoadPrefabMessage = "Loading with Prefab";
+        public string LoadProgressMessage = "Loading with Progress";
+        public string LoadProgressBarMessage = "Loading with Bar";
+        public string FinishMessage = "Finished!";
+
+        private void Start()
+        {
+            // Optimize the content for immersive headset
+            if (HolographicSettings.IsDisplayOpaque)
+            {
+               GameObject buttonCollection = GameObject.Find("ButtonCollection");
```

Comment:
Where is this button in relation to the GameObject this Component is on?

Reply:
The buttons are under ButtonCollection object in the scene hierarchy.
![2018-02-26 17_17_49-unity 2017 2 1f1 personal 64bit - progressexample unity - mixedrealitytoolkit-](https://user-images.githubusercontent.com/13754172/36705281-26c46442-1b19-11e8-8601-e0eda24d0f00.png)

Reply:
Okay, but where is the `ProgressExamples` component location in relation to this GameObject?

Reply:
i think the scaling of a scene object based on HMD type should be handled by a script that could be added to any UI element that differentially scales. There are two scenes (dialog and progress that use them)That way the object is always known by the instance and doesn't have to be searched for.